packages/ppx/src/generate_lib.re

```diff
@@ -136,32 +136,15 @@ let applyIgnore = (~loc, expr) => {
   );
 };
 
-/* createVariadicElement("div", newProps) */
-let variadicElement = (~loc, ~htmlTag) => {
+/* createVariadicElement(finalHtmlTag, newProps) */
+let variadicElement = (~loc) => {
   Helper.Exp.apply(
     ~loc,
     Helper.Exp.ident(~loc, withLoc(Lident("createVariadicElement"), ~loc)),
     [
       (
         Nolabel,
-        Helper.Exp.constant(
-          ~loc,
-          ~attrs=
-            Platform_attributes.rawLiteral(
-              ~loc,
-              Helper.Str.mk(
-                ~loc,
-                Pstr_eval(
-                  Helper.Exp.constant(
-                    ~loc,
-                    Pconst_string(htmlTag, loc, None),
-                  ),
-                  [],
-                ),
-              ),
-            ),
-          Pconst_string(htmlTag, loc, None),
-        ),
+        Helper.Exp.ident(~loc, withLoc(Lident("finalHtmlTag"), ~loc)),
```

Comment:
Maybe we can inline the switch expression here, that way we don't rely on anything outside of the createVariadicElement